src/diffusers/pipelines/consistency_models/pipeline_consistency_decoder.py

```diff
@@ -0,0 +1,135 @@
+
+        self.image_processor = VaeImageProcessor()
+
+    @torch.no_grad()
+    @replace_example_docstring(EXAMPLE_DOC_STRING)
+    def __call__(
+        self,
+        z: torch.Tensor,
+        generator: Optional[torch.Generator] = None,
+        output_type: Optional[str] = "pil",
+        num_inference_steps: int = 2,
+        return_dict: bool = True,
+    ):
+        r"""
+        Args:
+            z (`torch.Tensor`):
+                Latents which are typically precomputed from a reverse diffusion denoiser or an encoder.
+            num_inference_steps (`int`, *optional*, defaults to 2):
+                The number of denoising steps. More denoising steps usually lead to a higher quality image at the
+                expense of slower inference. Currently it doesn't support a quantity greater than 2.
+            generator (`torch.Generator`, *optional*):
+                A [`torch.Generator`](https://pytorch.org/docs/stable/generated/torch.Generator.html) to make
+                generation deterministic.
+            output_type (`str`, *optional*, defaults to `"pil"`):
+                The output format of the generated image. Choose between `PIL.Image` or `np.array`.
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether or not to return a [`~pipelines.ImagePipelineOutput`] instead of a plain tuple.
+
+        Examples:
+
+        Returns:
+            [`~pipelines.ImagePipelineOutput`] or `tuple`:
+                If `return_dict` is `True`, [`~pipelines.ImagePipelineOutput`] is returned, otherwise a `tuple` is
+                returned where the first element is a list with the generated images.
+        """
+        means = torch.tensor([0.38862467, 0.02253063, 0.07381133, -0.0171294], device=z.device, dtype=z.dtype)[
+            None, :, None, None
+        ]
+        stds = torch.tensor([0.9654121, 1.0440036, 0.76147926, 0.77022034], device=z.device, dtype=z.dtype)[
+            None, :, None, None
+        ]
+        z = (z - means) / stds
+        z = F.interpolate(z, mode="nearest", scale_factor=8)
+
+        batch_size, _, height, width = z.shape
+
+        self.scheduler.set_timesteps(num_inference_steps, device=self._execution_device)
+
+        x_t = self.scheduler.init_noise_sigma * randn_tensor(
+            (batch_size, 3, height, width), generator=generator, dtype=z.dtype, device=z.device
+        )
+
+        with self.progress_bar(total=num_inference_steps) as progress_bar:
+            for t in self.scheduler.timesteps:
+                model_input = torch.concat([self.scheduler.scale_model_input(x_t, t), z], dim=1)
+                model_output = self.unet(model_input, t).sample[:, :3, :, :]
+                prev_sample = self.scheduler.step(model_output, t, x_t, generator).prev_sample
+                x_t = prev_sample
+                progress_bar.update()
+
+        x_0 = x_t
+
+        image = self.image_processor.postprocess(x_0, output_type=output_type)
```

Comment:
Why not just pass `x_t` to postprocess directly?

Reply:
the variable rename is for documentation that at the end of the denoising loop we have t==0